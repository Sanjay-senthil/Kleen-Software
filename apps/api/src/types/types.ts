import {
  AccessControlPermission,
  Axis,
  EntityList,
  FilterQuery,
  GetListingDataResults,
  GetWidgetDataResult,
  ListItemBase as ListItem,
  ListingFormatItem,
  ListingItem,
  MultiTransFormationResults,
  Transformation,
  WorkflowFilter,
} from '@kleeen/types';

// Duplicated type, must unify in types library
export interface Attribute {
  aggregation?: Transformation;
  name: string;
  // TODO @carreta remove this when XORs can be aggregated [KSE3-1735]
  isXor?: boolean;
  hasMany?: boolean;
  formatType?: string;
}

export interface WorkflowFiltersResults {
  filters: WorkflowFilter[];
}

interface ActionItem {
  link?: string;
  title?: string;
  type?: string;
}

// TODO: @cafe move this and other shared types to @kleeen/types
export interface DispatchCustomActionResults {
  data: {
    actions?: ActionItem[];
    context?: any;
    customMessage?: string;
    customTitle?: string;
    functionName: string;
    success: boolean;
  };
}

export interface MultiTransFormationArgs {
  attributes?: string[];
  entity: string;
  filters: FilterQuery;
  transformations: Transformation[];
}

export interface FormatCheckResults {
  isValid: boolean;
  errors?: {
    message: string;
  }[];
}

export interface AccessControlCheckResults {
  accessLevel: AccessControlPermission;
}

export interface AuthContext {
  dataSources: Record<string, any>;
  headers: Record<string, any>;
  user: {
    email: string;
    'cognito:id': string;
  };
  token: string;
}

export {
  Axis,
  EntityList,
  GetListingDataResults,
  GetWidgetDataResult,
  ListingFormatItem,
  ListingItem,
  ListItem,
  MultiTransFormationResults,
  Transformation,
};
