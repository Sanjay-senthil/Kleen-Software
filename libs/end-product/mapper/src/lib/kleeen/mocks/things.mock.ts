import { ActionType, StatisticalDataType, ThingMap } from '@kleeen/types';

export const thingsMock: ThingMap = {
  completedService: {
    actions: [
      {
        areYouSure: true,
        component: 'CompletedServiceReportIssueModal',
        description: undefined,
        displayName: `Report Issue`,
        id: `4d882b7c-3773-443f-9327-16d168155e3c`,
        isCustomModal: true,
        name: `reportIssue`,
        type: ActionType.Custom,
      },
    ],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `dateScheduled`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecordType`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `downtimeHours`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `estimatedCost`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `datePerformed`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceCost`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceDescription`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `uuid`,
    deepDataType: undefined,
    description: ``,
    displayName: `CompletedService`,
    editable: false,
    format: { examples: [] },
    formatType: `uuid`,
    id: 59317,
    multiple: true,
    name: `completedService`,
    prototypeId: 59314,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.FreeForm,
    type: `uuid`,
  },
  currentMileage: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `integer`,
    deepDataType: undefined,
    description: ``,
    displayName: `CurrentMileage`,
    editable: false,
    format: { min: 0, max: 189000, examples: [] },
    formatType: `integer`,
    id: 59309,
    multiple: true,
    name: `currentMileage`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.NumericDiscrete,
    type: `integer`,
  },
  datePerformed: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `date`,
    deepDataType: undefined,
    description: ``,
    displayName: `DatePerformed`,
    editable: false,
    format: { examples: [] },
    formatType: `date`,
    id: 59303,
    multiple: true,
    name: `datePerformed`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.NumericTemporal,
    type: `date`,
  },
  dateScheduled: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `date`,
    deepDataType: undefined,
    description: ``,
    displayName: `DateScheduled`,
    editable: false,
    format: { examples: [] },
    formatType: `date`,
    id: 59308,
    multiple: true,
    name: `dateScheduled`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.NumericTemporal,
    type: `date`,
  },
  downtimeHours: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `integer`,
    deepDataType: undefined,
    description: ``,
    displayName: `DowntimeHours`,
    editable: false,
    format: { min: 1, max: 24, examples: [] },
    formatType: `integer`,
    id: 59305,
    multiple: true,
    name: `downtimeHours`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.NumericDiscrete,
    type: `integer`,
  },
  estimatedCost: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `integer`,
    deepDataType: undefined,
    description: ``,
    displayName: `EstimatedCost`,
    editable: false,
    format: { min: 1, max: 100, prefix: '$', suffix: 'k', examples: [] },
    formatType: `USD`,
    id: 59304,
    multiple: true,
    name: `estimatedCost`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.NumericDiscrete,
    type: `integer`,
  },
  scheduledService: {
    actions: [
      {
        areYouSure: true,
        component: 'ScheduledServiceReportIssueModal',
        description: undefined,
        displayName: `Report Issue`,
        id: `4d882b7c-3773-443f-9327-16d168155e3c`,
        isCustomModal: true,
        name: `reportIssue`,
        type: ActionType.Custom,
      },
    ],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `dateScheduled`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecordType`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `downtimeHours`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `estimatedCost`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `datePerformed`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceCost`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceDescription`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `uuid`,
    deepDataType: undefined,
    description: ``,
    displayName: `ScheduledService`,
    editable: false,
    format: { examples: [] },
    formatType: `uuid`,
    id: 59315,
    multiple: true,
    name: `scheduledService`,
    prototypeId: 59314,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.FreeForm,
    type: `uuid`,
  },
  serviceCost: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `integer`,
    deepDataType: undefined,
    description: ``,
    displayName: `ServiceCost`,
    editable: false,
    format: { min: 1, max: 100, prefix: '$', suffix: 'k', examples: [] },
    formatType: `USD`,
    id: 59307,
    multiple: true,
    name: `serviceCost`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.NumericDiscrete,
    type: `integer`,
  },
  serviceDescription: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `ServiceDescription`,
    editable: false,
    format: {
      examples: [
        'Flat Tire',
        'Broken Glass',
        'Oil Change',
        'Regular Service Interval',
        'Batttery',
        'Exterior Damage',
        'Engine Noise',
        'Suspension',
        'Brakes',
        'Electrical',
        'Interior',
        'Air Conditioning',
        'Navigation',
        'Error Code',
        'Emissions',
        'Inspection',
        'Accident Repair',
        'Transmission',
        'Performance',
        'Detailing',
      ],
    },
    formatType: `prime`,
    id: 59302,
    multiple: true,
    name: `serviceDescription`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  serviceRecord: {
    actions: [
      {
        areYouSure: true,
        component: 'ServiceRecordReportIssueModal',
        description: undefined,
        displayName: `Report Issue`,
        id: `4d882b7c-3773-443f-9327-16d168155e3c`,
        isCustomModal: true,
        name: `reportIssue`,
        type: ActionType.Custom,
      },
    ],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `dateScheduled`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecordType`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `downtimeHours`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `estimatedCost`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `datePerformed`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceCost`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceDescription`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `uuid`,
    deepDataType: undefined,
    description: ``,
    displayName: `ServiceRecord`,
    editable: false,
    format: { examples: [] },
    formatType: `uuid`,
    id: 59314,
    multiple: true,
    name: `serviceRecord`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.FreeForm,
    type: `uuid`,
  },
  serviceRecordType: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `serviceRecord`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `severity_level`,
    deepDataType: undefined,
    description: ``,
    displayName: `ServiceRecordType`,
    editable: false,
    format: {
      min: 1,
      max: 4,
      examples: ['Repair', 'Warranty', 'Recall', 'Preventative'],
      severityBad: 1,
      severityGood: 4,
      severityLevels: 4,
    },
    formatType: `severity_level`,
    id: 59313,
    multiple: true,
    name: `serviceRecordType`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.OrderedSeverityRanking,
    type: `severity_level`,
  },
  timestamp: {
    actions: [],
    attributes: [],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `timestamp`,
    deepDataType: undefined,
    description: undefined,
    displayName: `Timestamp`,
    editable: false,
    format: {},
    formatType: `timestamp`,
    id: 59271,
    multiple: true,
    name: `timestamp`,
    prototypeId: undefined,
    settings: {},
    statisticalType: StatisticalDataType.NumericTemporal,
    type: `timestamp`,
  },
  vehicle: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicleModel`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `serviceRecord`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicleBrand`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `currentMileage`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `warrantyStatus`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicleWarranty`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `scheduledService`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `completedService`,
      },
    ],
    canAddValues: false,
    canEditValues: true,
    crossLinking: [{ title: 'Vehicle Detail', slug: 'vehicle-detail' }],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `Vehicle`,
    editable: true,
    format: {
      examples: [
        'Fleet Car 1',
        'Fleet Car 2',
        'Fleet Car 3',
        'Fleet Car 4',
        'Loaner Car',
        'Loaner Van',
        'Loaner Truck',
        'Service Van 1',
        'Service Van 2',
        'Delivery Van',
        'Delivery Truck',
        'Service Truck 1',
        'Service Truck 2',
        'Executive Car 1',
        'Executive Car 2',
        'Executive Car 3',
        'Sales Car 1',
        'Sales Car 2',
        'Sales Car 3',
      ],
    },
    formatType: `prime`,
    id: 59333,
    multiple: true,
    name: `vehicle`,
    prototypeId: undefined,
    settings: { isEditable: true, isFilledByEU: true },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  vehicleBrand: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `vehicle`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleBrand`,
    editable: false,
    format: {
      examples: [
        'Datsun',
        'Delorean',
        'Pontiac',
        'Plymouth',
        'Desoto',
        'Mercury',
        'Tucker',
        'Oldsmobile',
        'Studebaker',
        'Kaiser',
        'Saturn',
        'AMC',
        'Edsel',
        'Auburn',
        'Checker',
        'Dusenburg',
        'Fisker',
        'Hudson',
        'MGB',
        'Triumph',
      ],
    },
    formatType: `prime`,
    id: 59330,
    multiple: true,
    name: `vehicleBrand`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  vehicleDtc: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicleDtcTimestamp`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `vehicleSoftwareVersion`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [{ title: 'DTC Detail', slug: 'dtc-detail' }],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleDtc`,
    editable: false,
    format: {
      examples: [
        'Air Cleaning Module Overcurrent - Circuit Current Above Threshold',
        'Backlite Defrost Relay Fault - Signal Shape / Waveform Failure',
        'Cabin Heater (PTCHF) 12V Logic Over Voltage - Circuit Voltage Above Threshold',
        'Door Handle Rear Left 12V Supply Under Voltage - Circuit Voltage Below Threshold',
        'Ext Door Handle Power RR Undercurrent - Circuit Current Below Threshold',
        'External 12VA Out of Range - Circuit Voltage Out of Range',
      ],
    },
    formatType: `prime`,
    id: 59726,
    multiple: true,
    name: `vehicleDtc`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  vehicleEcu: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicleSoftwareVersion`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleEcu`,
    editable: false,
    format: { examples: ['RIDM', 'BPCU', 'FIDM', 'RBCM', 'FBCM', 'DCAM', 'DCBM', 'OBCMD', 'OBCMP'] },
    formatType: `prime`,
    id: 48636,
    multiple: true,
    name: `vehicleEcu`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: true },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  vehicleModel: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `vehicle`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleModel`,
    editable: false,
    format: {
      examples: [
        'M1',
        'M2',
        'M3',
        'M5',
        'M4',
        'S1',
        'S2',
        'S3',
        'S4',
        'S5',
        'X1',
        'X2',
        'X3',
        'X4',
        'X5',
        'C1',
        'C2',
        'C3',
        'C4',
        'C5',
      ],
    },
    formatType: `prime`,
    id: 59328,
    multiple: true,
    name: `vehicleModel`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  vehicleSoftwareVersion: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicleEcu`,
      },
      {
        hasMany: true,
        isNullable: true,
        name: `vehicleDtc`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `vehicle`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleSoftwareVersion`,
    editable: false,
    format: { examples: ['5.4.0', '5.4.1', '5.4.2', '5.5.0', '6.0.0'] },
    formatType: `prime`,
    id: 59725,
    multiple: true,
    name: `vehicleSoftwareVersion`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: true },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  vehicleWarranty: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `warrantyStatus`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `uuid`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleWarranty`,
    editable: false,
    format: { examples: [] },
    formatType: `uuid`,
    id: 59320,
    multiple: true,
    name: `vehicleWarranty`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.FreeForm,
    type: `uuid`,
  },
  vehicleStateOfCharge: {
    actions: [],
    attributes: [
      {
        hasMany: true,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
    ],
    canAddValues: false,
    canEditValues: true,
    crossLinking: [],
    dataType: `severity_score`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleStateOfCharge`,
    editable: true,
    format: { min: 0, max: 100, severityBad: 0, severityGood: 100, severityLevels: 5 },
    formatType: `severity_score`,
    id: 48640,
    multiple: true,
    name: `vehicleStateOfCharge`,
    prototypeId: undefined,
    settings: { isEditable: true, isFilledByEU: true },
    statisticalType: StatisticalDataType.NumericSeverityRanking,
    type: `severity_score`,
  },
  vehicleTimeOfLastCharge: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `vehicle`,
      },
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `string`,
    deepDataType: undefined,
    description: ``,
    displayName: `VehicleTimeOfLastCharge`,
    editable: false,
    format: { examples: ['02/23/2022, 4:28:57 PM', '05/23/2022, 4:28:57 PM'] },
    formatType: `prime`,
    id: 48633,
    multiple: true,
    name: `vehicleTimeOfLastCharge`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: true },
    statisticalType: StatisticalDataType.Categorical,
    type: `string`,
  },
  warrantyStatus: {
    actions: [],
    attributes: [
      {
        hasMany: false,
        isNullable: false,
        name: `timestamp`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `vehicleWarranty`,
      },
      {
        hasMany: true,
        isNullable: false,
        name: `vehicle`,
      },
    ],
    canAddValues: false,
    canEditValues: false,
    crossLinking: [],
    dataType: `severity_level`,
    deepDataType: undefined,
    description: ``,
    displayName: `WarrantyStatus`,
    editable: false,
    format: {
      min: 1,
      max: 2,
      examples: ['Expired', 'Active'],
      severityBad: 1,
      severityGood: 2,
      severityLevels: 2,
    },
    formatType: `severity_level`,
    id: 59319,
    multiple: true,
    name: `warrantyStatus`,
    prototypeId: undefined,
    settings: { isEditable: false, isFilledByEU: false },
    statisticalType: StatisticalDataType.OrderedSeverityRanking,
    type: `severity_level`,
  },
};
