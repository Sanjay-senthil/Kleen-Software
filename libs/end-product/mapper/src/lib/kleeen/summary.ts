import { ColumnMap, GetListingDataResults } from '@kleeen/types';

import { getThingByName } from '@kleeen/things';
import { mapListColumns } from './list';

export function mapToSummary(rows: any[], columns: ColumnMap[]): GetListingDataResults {
  const [data] = mapListColumns(rows, columns, false);
  const reducedFormats = columns.reduce(
    (acc, ac) => ({ ...acc, [ac.key]: getThingByName(ac.key)?.format }),
    {},
  );
  return {
    data,
    format: reducedFormats,
  };
}
