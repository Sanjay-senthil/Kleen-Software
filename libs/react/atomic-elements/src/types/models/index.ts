import { CrossLinking, FormatProps, ResultsProps } from '@kleeen/types';

export interface ContextProps {
  data: DataProps;
  error: string;
  isLoading: boolean;
}

export interface DataProps extends ResultsProps {
  crossLinking: CrossLinking[] | CrossLinking;
  format: FormatProps;
}

export interface XAxisOptions extends Highcharts.XAxisOptions {
  categories?: [string];
  key?: string;
  valueLabels?: { label: string; value: number }[];
}

export interface YAxisOptions extends Highcharts.YAxisOptions {
  categories?: [string];
  key?: string;
  valueLabels?: { label: string; value: number }[];
}
