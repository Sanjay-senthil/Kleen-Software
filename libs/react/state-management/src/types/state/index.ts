export * from './endUser';
export * from './globalAppErrors';
export * from './investigation';
