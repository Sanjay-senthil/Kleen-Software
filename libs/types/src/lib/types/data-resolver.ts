import {
  Bucket,
  CrossLinking,
  CrossLinkingMatrix,
  DisplayValue,
  Transformation,
  UpdateDataStrategy,
} from '@kleeen/types';

/**
 * Enumerations
 */

export enum MapFunctionType {
  ToDisplayValue = 'toDisplayValue',
  ToChipsValue = 'toChipsValue',
  ToIdValue = 'toIdValue',
}

/**
 * Interfaces
 */

export interface Axis {
  categories?: string[] | ListItemBase[];
  key?: string;
  type?: string;
}

export interface ColumnMap {
  handler?: MapFunctionType;
  idPath?: string;
  key: string;
  path: string;
  showDisplayMedia?: boolean;
}

export interface Entity {
  name: string;
  properties: { [property: string]: { statisticalType: string } };
}

export interface EntityList {
  [key: string]: ListingItem | ListingItem[];
}

export interface DataAggregationArgsDataPoint {
  bucket?: Bucket;
  name: string;
  transformation?: Transformation;
  transformations?: Transformation[];
}

export interface GetListingDataResults {
  data: EntityList[];
  format: { [key: string]: ListingFormatItem };
  latestRequestTimestamp?: number;
  pagination?: { totalCount: number };
  strategy?: UpdateDataStrategy;
}

export interface GetWidgetDataResult {
  crossLinking?: CrossLinkingMatrix;
  format: {
    xAxis?: Axis;
    yAxis?: Axis;
  };
  results: number[] | number[][];
}

export interface ListingFormatItem {
  aggregations?: null;
  examples?: null;
  max?: null;
  min?: null;
  prefix?: null;
  severityBad?: null;
  severityGood?: null;
  severityLevels?: null;
  suffix?: null;
}

export interface ListingItem {
  displayValue: DisplayValue;
  id?: string;
}

export interface MultiTransFormationResults {
  crossLinking?: CrossLinking[];
  format: {
    xAxis?: Axis;
    yAxis?: Axis;
  };
  results: number | number[] | number[][];
  transformation: Transformation;
}

export interface TransformationArgs extends Pick<ColumnMap, 'idPath' | 'key' | 'path' | 'showDisplayMedia'> {
  row: any;
}

/**
 * Types
 */

export type ListItemBase = number | string | boolean | undefined;
