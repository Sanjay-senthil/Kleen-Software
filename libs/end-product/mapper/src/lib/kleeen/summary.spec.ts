import { Maybe, Thing } from '@kleeen/types';

import { mapToSummary } from './summary';
import { thingsMock } from './mocks';

jest.mock('@kleeen/things', () => {
  return {
    getThingByName(name: string): Maybe<Thing> {
      return Object.values(thingsMock).find((thing) => thing.name === name) as Thing;
    },
  };
});

describe('kleeen', () => {
  describe('summary', () => {
    it('should return "GetListingDataResults" response if "mapToSummary" is called from a Summary Entity Detail widget', () => {
      const rows = [
        {
          'Vehicle.vehicleId': 7,
          'Vehicle.brand': 'BYD',
          'Vehicle.model': 'S2',
          'Vehicle.warrantyStatus': 'Active',
          'Service.completedCount': 10,
          'Service.scheduledCount': 1,
          'Vehicle.currentMileage': 222206,
        },
      ];
      const columns = [
        {
          idPath: 'Vehicle.vehicleId',
          key: 'vehicle',
          path: 'Vehicle.vehicleId',
        },
        { key: 'vehicleBrand', path: 'Vehicle.brand' },
        { key: 'vehicleModel', path: 'Vehicle.model' },
        { key: 'vehicleWarranty', path: 'Vehicle.warrantyStatus' },
        { key: 'completedService', path: 'Service.completedCount' },
        { key: 'scheduledService', path: 'Service.scheduledCount' },
        { key: 'currentMileage', path: 'Vehicle.currentMileage' },
      ];

      expect(mapToSummary(rows, columns)).toMatchSnapshot();
    });
  });
});
