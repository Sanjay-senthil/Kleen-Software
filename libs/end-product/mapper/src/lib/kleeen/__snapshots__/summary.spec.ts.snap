// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kleeen summary should return "GetListingDataResults" response if "mapToSummary" is called from a Summary Entity Detail widget 1`] = `
Object {
  "data": Array [
    Object {
      "vehicle": Object {
        "displayValue": 7,
        "id": 7,
      },
    },
    Object {
      "vehicleBrand": Object {
        "displayMedia": Object {
          "type": "text",
          "value": "BYD",
        },
        "displayValue": "BYD",
        "id": undefined,
      },
    },
    Object {
      "vehicleModel": Object {
        "displayMedia": Object {
          "type": "text",
          "value": "S2",
        },
        "displayValue": "S2",
        "id": undefined,
      },
    },
    Object {
      "vehicleWarranty": Object {
        "displayMedia": null,
        "displayValue": "Active",
        "id": undefined,
      },
    },
    Object {
      "completedService": Object {
        "displayValue": 10,
        "id": undefined,
      },
    },
    Object {
      "scheduledService": Object {
        "displayValue": 1,
        "id": undefined,
      },
    },
    Object {
      "currentMileage": Object {
        "displayValue": 222206,
        "id": undefined,
      },
    },
  ],
  "format": Object {
    "completedService": Object {
      "examples": Array [],
    },
    "currentMileage": Object {
      "examples": Array [],
      "max": 189000,
      "min": 0,
    },
    "scheduledService": Object {
      "examples": Array [],
    },
    "vehicle": Object {
      "examples": Array [
        "Fleet Car 1",
        "Fleet Car 2",
        "Fleet Car 3",
        "Fleet Car 4",
        "Loaner Car",
        "Loaner Van",
        "Loaner Truck",
        "Service Van 1",
        "Service Van 2",
        "Delivery Van",
        "Delivery Truck",
        "Service Truck 1",
        "Service Truck 2",
        "Executive Car 1",
        "Executive Car 2",
        "Executive Car 3",
        "Sales Car 1",
        "Sales Car 2",
        "Sales Car 3",
      ],
    },
    "vehicleBrand": Object {
      "examples": Array [
        "Datsun",
        "Delorean",
        "Pontiac",
        "Plymouth",
        "Desoto",
        "Mercury",
        "Tucker",
        "Oldsmobile",
        "Studebaker",
        "Kaiser",
        "Saturn",
        "AMC",
        "Edsel",
        "Auburn",
        "Checker",
        "Dusenburg",
        "Fisker",
        "Hudson",
        "MGB",
        "Triumph",
      ],
    },
    "vehicleModel": Object {
      "examples": Array [
        "M1",
        "M2",
        "M3",
        "M5",
        "M4",
        "S1",
        "S2",
        "S3",
        "S4",
        "S5",
        "X1",
        "X2",
        "X3",
        "X4",
        "X5",
        "C1",
        "C2",
        "C3",
        "C4",
        "C5",
      ],
    },
    "vehicleWarranty": Object {
      "examples": Array [],
    },
  },
}
`;
